### ISSUE TITLE
IPv6 Support

### ISSUE THREAD

**Issue by genofire** [opendata-stuttgart/sensors-software#605]
We run already a ipv6only network (with dns64 and nat64 - for ipv4only services in the internet).

**Comment by dirkmueller** [comment]
First step would be to compile in IPv6 support (which is currently disabled due to size constraints)

**Comment by genofire** [comment]
Do you have a hint, where i could find the build flags to try it on my own?

**Comment by dirkmueller** [comment]
In the platformio.ini you find the lwip variant as a -D define. We currently only build with ipv4.

I tried getting ipv6 set up however my provider does not support it..

See https://docs.platformio.org/en/latest/platforms/espressif8266.html#lwip-variant for the possible build flags

**Comment by dirkmueller** [comment]
Ist nun in B2 so gesetzt. Getestet habe ich noch nichts

**Comment by genofire** [comment]
German ^^: nice, werde ich am Dienstag ausprobieren xD

**Comment by genofire** [comment]
So, er ist nun im DualStack Netzwerk erreichbar :+1:  im IPv6only Netz keine Ahnung.

Frag mich gerade, wie man ggf. die Software etwas härten könnte:
- HTTPS (selfsigning oder Let's Encrypt)
- IP-Subnet white-/blacklist als Firewall oder im "Webservice"

**Comment by dirkmueller** [comment]
@genofire yeah, there are security implications. I am not a big fan of https webserver right now because the way Arduino core implements this is with very high overhead. I have lost a bit of confidence on being able to fix that quickly.

We likely need to restrict to link local addresses for now, everything else is too dangerous. Does the mdns/dns-sd resolve to ipv6? To ipv4? Both?

**Comment by genofire** [comment]
please do not restrict to lladdr - so it's not reachable by e.g. firefox or other tools.
maybe restrict to:
- lladdr and given subnet 
- or a whitelist
- or let it like current state (most router like AVM and so on has a good firewall for such things) 

--- i do not use this features, but it will try mdns / avahi next thuesday

**Comment by dirkmueller** [comment]
Okay. Does the user have to configure the whitelist prefix or can we just learn it via slaac?

I think we don't have enough insights on how sensors are being deployed to know whether we're safely able to fix on good filtering at the perimeter.

Given the gaping holes in the UI (Don't want to be more explicit until fixed) we do need to be careful with this.

Please advice what the minimum steps are to make this safe. If I can't get that implemented timely I will have to turn off ipv6 for a stable release until we had time to do basic things right.

**Comment by genofire** [comment]
for the first release i would learn it by "route advertisement" (RA) (which is used as subnet of configurated by slaac) and maybe a checkbox to disable this firewall.

in 2. step a textarea in the UI will be nice, where the RA subnet is already in in format like:
```
fe80::/64
2001:db8:1:2::/64
```
---
but like i say, that is not a blocking think (a ipv6 network scan to find such devices, is not easy - on most private networks impossible thanks "good" routers firewall like the fritzbox)
